src/test/java/uk/gov/pay/ledger/util/pagination/PaginationBuilderTest.java

```diff
@@ -138,4 +141,27 @@ public void shouldNotGenerateNextLink_whenLimitTotalIsSetAndCountIsLessThanPageS
         builder = builder.buildResponse();
         assertThat(builder.getNextLink(), is(nullValue()));
     }
-}
\ No newline at end of file
+    
+    @Test
+    public void shouldGenerateSelfLinkWithOnlyPrescribedSpecialCharactersURLEncoded(){
+        Map<String, Object> valuesForReferenceParameter = Map.of(
+                "{ref", "reference=%7Bref",
+                "[ref", "reference=%5Bref",
+                "r{e{f}{", "reference=r%7Be%7Bf%7D%7B",
+                "ref=", "reference=ref=",
+                "ref&", "reference=ref&",
```

Comment:
Having a test for a value like `foo=bar&baz=quux` (as a single value) and checking it’s not tokenised into multiple values would be useful

Reply:
Gotcha 👌!